neptune-core/src/application/loops/main_loop.rs

```diff
@@ -2066,6 +2066,30 @@ impl MainLoopHandler {
                 // shut down
                 Ok(true)
             }
+            RPCServerToMain::SubmitTx(transaction) => {
+                // Technically RPC is not supposed/capable to handle ProofWitness but just in case
+                // Until a decision is made. (As this might get used by api too.)
+                let notification: Option<TransactionNotification> =
+                    transaction.as_ref().try_into().ok();
+
+                {
+                    let mut state = self.global_state_lock.lock_guard_mut().await;
+
+                    state
+                        .mempool_insert(*transaction, UpgradePriority::Critical)
+                        .await;
```

Comment:
I wonder if `Critical` is the right value to use here. Let's keep it for now. But it will mean that this node will put in a big effort into upgrading the proof of the transaction. Perhaps we'd eventually want an endpoint to submit transactions where the node calculates its economic interest in a transaction and then uses this to ascribe a priority? An endpoint similar to that of [`incentivized_proof_collection_transaction` on the wallet-server branch](https://github.com/Neptune-Crypto/neptune-core/blob/ef59f1b0b9e4e41f807f8b68c245ae7fd51cc50f/neptune-core/src/application/rest_server/mod.rs#L483C10-L483C51).